### DISCUSSION TITLE
Compare formatting styles

### DISCUSSION THREAD

**Discussion by xixiaofinland** [xixiaofinland/afmt#24] [General]
Some styling differences I found.

**Comment by xixiaofinland** [comment]
A
```
class SafeNavigationOperator {
  void simpleOperator() {
    Account c =
      [SELECT Id FROM Account WHERE Id = '001000000FAKEID'] ??
      [
        SELECT Id
        FROM Account
        WHERE Id = '001000001FAKEID' AND Very_Long_Field_Name__c = 'Hello'
      ]
    ;
  }
}
```

B:
```
class SafeNavigationOperator {
  void simpleOperator() {
    Account c = [SELECT Id FROM Account WHERE Id = '001000000FAKEID'] ??
      [
        SELECT Id
        FROM Account
        WHERE Id = '001000001FAKEID' AND Very_Long_Field_Name__c = 'Hello'
      ];
  }
}
```

**Reply by aheber** [reply]
As a note, I think this is the Null Coalescing Operator rather than the safe navigation:
https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_NullCoalescingOperator.htm

I think I prefer `B` but I'm sure I could find code where that wouldn't always be true. I might even change my preference depending on if the first portion needed to be wrapped. If the first SOQL query was long enough to force wrapping should the entire query start on the next line to provide more clear delineation?

**Reply by pkozuchowski** [reply]
I'd expect it to be...both :D 
If I left new line after assignment operator, then that should be respected.

But I could also do something like `[SELECT Id FROM Account]?.Id ?? DEFAULT_ACCOUNT;` - I probably wouldn't want that to be broken apart? Well, depends if the line is within line limit.
